src/core/thread/mle_router.cpp

```diff
@@ -4298,66 +4298,56 @@ bool MleRouter::HasMinDowngradeNeighborRouters(void)
 
 bool MleRouter::HasOneNeighborWithComparableConnectivity(const RouteTlv &aRoute, uint8_t aRouterId)
 {
-    bool rval = true;
+    uint8_t routerCount = 0;
+    bool    rval        = true;
 
     // process local neighbor routers
     for (Router &router : Get<RouterTable>().Iterate())
     {
-        uint8_t localLinkQuality = 0;
-        uint8_t peerLinkQuality  = 0;
-        uint8_t routerCount      = 0;
+        uint8_t localLinkQuality;
+        uint8_t peerLinkQuality;
 
-        if (router.GetRouterId() == mRouterId)
+        if (!router.IsStateValid() || (router.GetRouterId() == mRouterId) || (router.GetRouterId() == aRouterId))
```

Comment:
```suggestion
        if (!router.IsStateValid() || router.GetRouterId() == mRouterId) || (router.GetRouterId() == aRouterId)
```
seems redundant parentheses?